aegisthus-hadoop/src/main/java/com/netflix/aegisthus/util/ObservableToIterator.java

```diff
@@ -57,7 +62,9 @@ public boolean hasNext() {
                     buf = take();
                 }
                 if (buf.isOnError()) {
-                    throw Exceptions.propagate(buf.getThrowable());
```

Comment:
Do you have a stacktrace of the exception? It seems to me that logging it here vs in the application-layer code would yield the same exception (since Exceptions.propagate should just propagate `buf.getThrowable`)